src/translate.py

```diff
@@ -2741,16 +2765,20 @@ def function_abi(fn_sig: FunctionSignature, *, for_call: bool) -> Abi:
         only_floats = False
 
     for ind, param in enumerate(fn_sig.params):
-        size, align = param.type.get_size_align_bytes()
+        param_type = param.type
+        if param_type.is_array():
+            # Array arguments decay into pointers
+            param_type = param_type.reference()
+        size, align = param_type.get_size_align_bytes()
```

Comment:
we should rename get_size_align_bytes -> get_parameter_size_align_bytes btw, to ensure it's not used by accident (it contains a sketchy "or 32" that only makes sense for function calls, if even then)